src/main/java/net/sacredlabyrinth/phaed/simpleclans/managers/ChatManager.java

```diff
@@ -0,0 +1,175 @@
+                if (channel == Channel.ALLY) {
+                    receivers.addAll(clan.getOnlineAllyMembers().stream().filter(allymember -> !allymember.isMutedAlly()).collect(Collectors.toList()));
+                }
+
+            /*
+              TODO: Make it async, change Type to Channel
+              (and move it to another method, probably?)
+             */
+                new BukkitRunnable() {
+                    @Override
+                    public void run() {
+                        ChatEvent event = new ChatEvent(message, clanPlayer, receivers, ChatEvent.Type.ALLY);
+                        Bukkit.getServer().getPluginManager().callEvent(event);
+
+                        if (event.isCancelled()) {
+                            return;
+                        }
+
+                        String message = formatChat(channel, clanPlayer, event.getMessage(), event.getPlaceholders());
+                        String eyeMessage = formatSpyClanChat(clanPlayer, message);
+                        plugin.getLogger().info(message);
+
+                        for (ClanPlayer cp : receivers) {
+                            ChatBlock.sendMessage(cp, message);
+                        }
+
+                        sendToAllSeeing(eyeMessage, receivers);
+                    }
+                }.runTask(plugin);
+            }
+        }
+    }
+
+    private String formatChat(Channel channel, ClanPlayer cp, String msg, Map<String, String> placeholders) {
+        SettingsManager sm = plugin.getSettingsManager();
+
+        String leaderColor = sm.getClanChatLeaderColor();
+        String memberColor = sm.getClanChatMemberColor();
+        String trustedColor = sm.getClanChatTrustedColor();
+
+        if (channel == Channel.ALLY) {
+            leaderColor = sm.getAllyChatLeaderColor();
+            memberColor = sm.getAllyChatMemberColor();
+            trustedColor = sm.getClanChatTrustedColor();
+        }
+
+        String rank = cp.getRankId().isEmpty() ? null : ChatUtils.parseColors(cp.getRankDisplayName());
+        String rankFormat = rank != null ? ChatUtils.parseColors(sm.getClanChatRank()).replace("%rank%", rank) : "";
+
+        String message = replacePlaceholders(sm.getClanChatFormat(), cp, leaderColor, trustedColor, memberColor, rankFormat, msg);
+
+        if (placeholders != null) {
+            for (Map.Entry<String, String> e : placeholders.entrySet()) {
+                message = message.replace("%" + e.getKey() + "%", e.getValue());
+            }
+        }
+        return message;
+    }
+
+    private String formatSpyClanChat(ClanPlayer cp, String msg) {
+        msg = ChatUtils.stripColors(msg);
+
+        if (msg.contains(ChatUtils.stripColors(cp.getClan().getColorTag()))) {
+            return ChatColor.DARK_GRAY + msg;
+        } else {
+            return ChatColor.DARK_GRAY + "[" + cp.getTag() + "] " + msg;
+        }
+    }
```

Comment:
Currently, it formats not clan chat only, allies chat too.
It can be confusing for all-seeing-eyes.

So, to make this difference, maybe, we could add [A] - as in ally's chat and [C] - as in clan's chat. 🤔 
Probably, different colors of messages?

Reply:
Or a format fetched from config.